extensions/arc/src/extension.ts

```diff
@@ -54,6 +50,19 @@ export async function activate(context: vscode.ExtensionContext): Promise<void>
 	});
 
 	await checkArcDeploymentExtension();
+
+	return {
+		getAzureArcTreeDataProvider: () => treeDataProvider
```

Comment:
Don't expose the whole tree provider - that's unnecessary and has weird consequences with other things being able to control our tree.

Just have a function to return the currently registered controllers instead.

Reply:
so just an api to return treeDataProvider.getChildren()?

Reply:
and then later i will add an API to invoke the add/update controller dialog

Reply:
The dialog you can just use the existing command for - no need to add an api just for that.

And yes - the api here would be something like 

`getRegisteredDataControllers(): ControllerInfo[]` 

where ControllerInfo is https://github.com/microsoft/azuredatastudio/blob/main/extensions/arc/src/models/controllerModel.ts#L14 which is this : ControllerTreeNode.model.info

Reply:
done.